tests/models/fuyu/test_processing_fuyu.py

```diff
@@ -156,6 +172,159 @@ def test_fuyu_processing_multiple_image_sample(self):
         self.assertTrue((processor_outputs["image_patches_indices"] == expected_image_patch_inputs).all())
         self.assertTrue((processor_outputs["input_ids"] == expected_padded_unpacked_token_inputs).all())
 
+    @require_vision
+    @require_torch
+    def test_kwargs_overrides_default_tokenizer_kwargs(self):
+        if "image_processor" not in self.processor_class.attributes:
+            self.skipTest(f"image_processor attribute not present in {self.processor_class}")
+        image_processor = self.get_component("image_processor")
+        tokenizer = self.get_component("tokenizer", max_length=117)
+        if not tokenizer.pad_token:
+            tokenizer.pad_token = "[TEST_PAD]"
+
+        processor = self.processor_class(tokenizer=tokenizer, image_processor=image_processor)
+        self.skip_processor_without_typed_kwargs(processor)
+        input_str = "lower newer"
+        # Fuyu uses tokenizer kwargs only when image is None.
+        image_input = None
+
+        inputs = processor(
+            text=input_str, images=image_input, return_tensors="pt", max_length=112, padding="max_length"
+        )
+        self.assertEqual(len(inputs["input_ids"][0]), 112)
+
+    def test_image_processor_defaults_preserved_by_image_kwargs(self):
+        self.skipTest(reason="Fuyu processor does not support image_processor kwargs")
+
+    def test_kwargs_overrides_default_image_processor_kwargs(self):
+        self.skipTest(reason="Fuyu processor does not support image_processor kwargs")
+
+    @require_vision
+    @require_torch
+    def test_tokenizer_defaults_preserved_by_kwargs(self):
+        if "image_processor" not in self.processor_class.attributes:
+            self.skipTest(f"image_processor attribute not present in {self.processor_class}")
+        image_processor = self.get_component("image_processor")
+        tokenizer = self.get_component("tokenizer", max_length=117, padding="max_length")
+
+        if not tokenizer.pad_token:
+            tokenizer.pad_token = "[TEST_PAD]"
+
+        processor = self.processor_class(tokenizer=tokenizer, image_processor=image_processor)
+        self.skip_processor_without_typed_kwargs(processor)
+        input_str = "lower newer"
+        image_input = None
```

Comment:
curious, this line is the sole difference with the common test, is that correct? Why do we need to pass `None` here?

Reply:
Forgot to comment above this line, but in the current Fuyu processor, when both images and text are passed in, the processor doesn't allow for any tokenizer kwargs as the tokenization process is very specific to this model when working with images. So if `image_input `is not None, the tokenizer kwargs would just be ignored and the test won't pass. Kosmos-2 processor does the same.

Reply:
ah, I understand - thank you!